### PR TITLE
Created CICD pipeline.

### PR DESCRIPTION
**Pull request by joakim077**
Added guide to create CICD pipeline and configure nginx as a proxy server, also added SSL encryption for secure HTTPS communication.

## Release Notes

- **New Features**
  - Introduced environment variables for user identification and Docker image configuration.
  - Added a Jenkins pipeline for automated code checkout, image building, pushing, and deployment.
  - New section in the README for the CICD pipeline with a link to a detailed guide.
  - Comprehensive guide for configuring Nginx as a proxy server with SSL support.
  - Transitioned MySQL service to use named volumes for data persistence.

- **Documentation**
  - Enhanced `cicd.md` with detailed steps for setting up the CICD workflow.
  - Minor formatting adjustments in the README.

### CODE REVIEW

**Review by LondheShubham153 — CHANGES_REQUESTED** [review]
@joakim077 great work, 
there are a few suggestions from @coderabbit 
please add them as well

**Comment by joakim077** [comment]
@LondheShubham153 sure sir

File: cicd.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    http://<Instance_IP>:8080
+    ```
+    Login and install Suggested Plugins.
+
+2. #### Jenkins Configuration.
+    - SetUp Agent
+        - Agents are used for distribute the builds in parallel execution
+        ![Agent](images/agent.png)
+
+        - Install Docker and docker-compose:V2 on worker node and add the user who is executing the Jenkins job into the docker group. because we are going to deploy application in worker node itself.
```
[the remark this answers is not in the document]
Comment (joakim077) [reply cicd.md:26]: yes

File: cicd.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+3. #### Create a Pipeline, Execute Job and Configure Webhook
+ 
+    - Configure Pipeline.
+        - Configure job to get pipeline from SCM.
+        ![pipeline](images/pipeline.png)
+    - Build Job and Check
+        - Build the Job
+    - Configure WebHook and poll SCM.
+        - Webhook: Automatically built when a change is made to the Source code.
+        - Poll: periodically monitor the repository and if any changes are detected, build Job.
```
[the remark this answers is not in the document]
Comment (joakim077) [reply cicd.md]: yes

File: cicd.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
+#### CICD Workflow
+- Cloning the Project code from GitHub.
+- Build docker image and push it to docker hub.
+- Pull image from docker hub and deploy application using docker compose .
```
[the remark this answers is not in the document]
Comment (joakim077) [reply cicd.md:4]: provide a template

**Review by LondheShubham153 — APPROVED** [review]

**Pull request merged**